comp/host-profiler/collector/impl/collector.go

```diff
@@ -36,7 +33,8 @@ func NewParams(uri string) Params {
 
 // Requires defines the dependencies for the collector component
 type Requires struct {
-	Params Params
+	Params         Params
+	ExtraFactories ExtraFactories
```

Comment:
Is the new interface `ExtraFactories` required here in order to work with fx? If not, I think the code could be easier to follow if we add this logic to `newCollectorSettings` instead by passing `CoreConfPath` into `Params`, and check if this string is empty to determine what factories to include instead of creating `NewExtraFactoriesWithAgentCore` and `NewExtraFactoriesWithoutAgentCore`.

Reply:
Can you explain a bit more what you have in mind? 

The remote tagger needs a config and so `CoreConfPath` to be instanciated. It means you cannot easily pass a instance of the remote tagger `newCollectorSettings` and later decide to not use it.

Reply:
I see, I think that makes sense then to leave it as is